code-challenges.test.js

```diff
@@ -14,43 +14,105 @@
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const fibonacciLength1 = 6
-// Expected output: [1, 1, 2, 3, 5, 8]
-
-const fibonacciLength2 = 10
-// Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
+describe("fibonacciSequence", () => {
+  it("takes in a number (greater than 2) and returns an array containing the Fibonacci sequence", () => {
+    const fibonacciLength1 = 6
+    const fibonacciLength2 = 10
+    // Expected output: [1, 1, 2, 3, 5, 8]
+    let result1 =  [1, 1, 2, 3, 5, 8]
+    // Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
+    let result2 = [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
+    expect(fibonacciSequence(fibonacciLength1)).toEqual(result1)
+    expect(fibonacciSequence(fibonacciLength2)).toEqual(result2)
+  })
+})
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+// - Create a function named fibonacciSequence
+// - Input: number (length)
+// - Output: array containing the Fibonacci Sequence
+    // note: the length of the array is determined by the argument of the function
+    // 1. Use a condtional to check for base cases
+      // if length is less than 1 => empty array
+      // if length is 1 or 2 => array with the first one or two Fibonacci nums
+    // 2. Intialize the sequence array with the first two Fibonacci nums, 0 and 1
+    // 3. Use a for loop to calculate and push the next value to the sequence until the desired length is reached. 
+      // set the value to 2 
+      // create a next value variable to set the sequence
+      // return the sequence
+  
+
+    const fibonacciSequence = (num) => {
+      if(num < 1){
+        return [];
+      } else if(num === 1) {
+        return [0];
+      } else if(num === 2){
+        return [0, 1];
+      }
+
+      const sequence = [0, 1];
+      for(let i = 2; i < num; i++){
+        const nextValue = sequence[i - 1] + sequence[i - 2];
+        sequence.push(nextValue);
+      }
+      return sequence;
+    }
```

Comment:
Looking at the expected outputs, the sequence should start with 1, not 0. The test will fail because of this.
Here's how you could rewrite this pseudo code to make it more clear:
- create variable sequence holding first 2 values in the fibonacci pattern starting from 1
- iterate using for loop starting at index 2
- input number will determine how many times the loop iterates
- each iteration adds the values at the previous 2 indices and stores the sum in a variable nextValue
- nextValue is added to the sequence array

Reply:
Thank you so much this really helps!